app/src/main/java/org/piwigo/bg/UploadService.java

```diff
@@ -118,16 +121,21 @@ protected void onHandleIntent(Intent intent)
         snackProgressEvent.setSnackbarDuration(SnackProgressBarManager.LENGTH_INDEFINITE);
         snackProgressEvent.setAction(SnackProgressEvent.SnackbarUpdateAction.REFRESH);
         snackProgressEvent.setSnackbarDesc(getResources().getString(R.string.upload_started));
-        onUploadStarted(imageUploadQueue);
+        onChunkedUploadStarted(imageUploadQueue);
     }
 
-    protected void onUploadStarted(ImageUploadQueue<UploadAction> imageUploadQueue)
-    {
+    protected void onChunkedUploadStarted(ImageUploadQueue<UploadAction> imageUploadQueue) {
         final UploadAction uploadAction;
         if ((uploadAction = imageUploadQueue.poll()) == null)
             return;
+
+        //Chunks
+        long tmpFileSize = getFileSize(uploadAction.getUploadData().getTargetUri());
+        int chunkSize = getChunkSize(); // Chunk max size = 1MB
+        int expectedChunks = tmpFileSize < chunkSize ? 1 : (int) (tmpFileSize / chunkSize) + 1;
```

Comment:
does this work for images that have EXACTLY the junk size? I guess not...

Reply:
I guess we could go for a "<= " comparator

Reply:
This seems correct, yes.

Reply:
Fixed in https://github.com/Piwigo/Piwigo-Android/pull/214/commits/63cb939f931ed73f29d20f059ac6d7da0534dc14